packages/server/api/src/app/flows/issues/issues-service.ts

```diff
@@ -98,43 +169,74 @@ export const issuesService = (log: FastifyBaseLogger) => ({
         }, {
             status,
             updated: new Date().toISOString(),
-            count: 0,
         })
     },
     async update({ projectId, flowId, status }: {
         projectId: ApId
         flowId: ApId
         status: IssueStatus
     }): Promise<Issue> {
-        if (status != IssueStatus.RESOLEVED) {
-            await repo().increment({ projectId, flowId }, 'count', 1)
-        }
         await repo().update({
             projectId,
             flowId,
         }, {
-            ...spreadIfDefined('lastOccurrence', status !== IssueStatus.RESOLEVED ? dayjs().toISOString() : undefined),
-            ...spreadIfDefined('count', status === IssueStatus.RESOLEVED ? 0 : undefined),
+            ...spreadIfDefined('lastOccurrence', status !== IssueStatus.RESOLVED ? dayjs().toISOString() : undefined),
             status,
             updated: new Date().toISOString(),
         })
-        return repo().findOneByOrFail({
+        const issue = await repo().findOneByOrFail({
             projectId,
             flowId,
         })
+        if (!isNil(issue.stepName)) {
+            issue.step = await getStepFromFlow(issue.flowId, issue.stepName, log)
+        }
+
+        return issue
     },
     async count({ projectId }: { projectId: ApId }): Promise<number> {
         return repo().count({
             where: {
                 projectId,
-                status: IssueStatus.ONGOING,
+                status: IssueStatus.UNRESOLVED,
             },
         })
     },
+    async archiveOldIssues(olderThanDays: number): Promise<void> {
+        const cutoffDate = dayjs().subtract(olderThanDays, 'days').toISOString()
+        
+        const result = await repo().update({
+            status: IssueStatus.UNRESOLVED,
+            updated: LessThan(cutoffDate),
+        }, {
+            status: IssueStatus.ARCHIVED,
+            updated: new Date().toISOString(),
+        })
+
+        log.info({
+            archivedCount: result.affected,
+            olderThanDays,
+            cutoffDate,
+        }, 'Archived old issues')
+    },
 })
 
 type UpdateParams = {
     projectId: string
     id: string
     status: IssueStatus
+}
+
+const getStepFromFlow = async (flowId: string, stepName: string, log: FastifyBaseLogger) => {
```

Comment:
You are fetching flowVersion twice, once in list and one here.

But if you have enrichIssue function that just takes issue and use it both in list and get it would be much better.